src/Backends/InterfaceSettings.vala

```diff
@@ -48,6 +59,19 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
                 sync_gsettings_to_accountsservice ();
             }
         });
+
+        background_settings.changed.connect ((key) => {
+            if (key == PICTURE_OPTIONS ||
+                key == PRIMARY_COLOR) {
+                sync_gsettings_to_accountsservice ();
+                return;
+            }
+
+            if (key == PICTURE_URI ||
+                key == PICTURE_URI_DARK) {
+                sync_background_to_greeter ();
+            }
```

Comment:
instead of a single function dealing with both cases, a `set_greeter_wallpaper(string uri, string name)` function would be simpler and cleaner. but given that we don't do anything with the dark wallpaper, i would keep it off from this.